src/main/java/frc/robot/commands/VarShootPrime.java

```diff
@@ -0,0 +1,103 @@
+import frc.robot.classes.Util;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Wrist;
+import frc.robot.subsystems.chassis.PoseEstimator;
+import java.util.function.Supplier;
+import org.littletonrobotics.junction.Logger;
+
+public class VarShootPrime extends Command {
+  private Wrist wrist;
+  private Shooter shooter;
+  private PoseEstimator poseEstimator;
+  private Supplier<Translation2d> speakerTranslation;
+
+  // Translation of where the note exits in the XZ plane (side view)
+  private final Translation2d shooterXZTrans;
+
+  private final Range velRange; // Range of velocity from min distance to max distance
+  private final Range distRange; // Range of distance from min distance to max distance
+  private final double heightLengthCoeff;
+  private final double RPM_MPS;
+  private final double wristStow;
+
+  /** Creates a new VarShootPrime. */
+  public VarShootPrime(
+      Wrist wrist,
+      Shooter shooter,
+      PoseEstimator poseEstimator,
+      Translation2d shooterXZTrans,
+      Range velRange,
+      Range distRange,
+      double thetaCoeff,
+      double RPM_MPS,
+      double wristStow) {
+    this.wrist = wrist;
+    this.shooter = shooter;
+    this.poseEstimator = poseEstimator;
+    this.speakerTranslation = Constants.SPEAKER_TRANSLATION;
+    this.shooterXZTrans = shooterXZTrans;
+    this.velRange = velRange;
+    this.distRange = distRange;
+    this.heightLengthCoeff = thetaCoeff;
+    this.RPM_MPS = RPM_MPS;
+    this.wristStow = wristStow;
+
+    addRequirements(wrist, shooter);
+  }
+
+  @Override
+  public void execute() {
+    Pose2d pose = poseEstimator.getFusedPose();
+
+    // Distance and height to speaker
+    double l = pose.getTranslation().getDistance(speakerTranslation.get()) - shooterXZTrans.getX();
+    double h = (Constants.SPEAKER_HEIGHT - shooterXZTrans.getY());
+    // Calculate velocity based on lerping within the velocity range based on the distance range
+    // double v = Util.lerp(Util.clamp(h, distRange) / distRange.getRange(), velRange);
+    double v = velRange.max;
+    double theta = calcTheta(Constants.GRAVITY, l, h, v);
+    theta = Units.radiansToDegrees(theta);
+    double modify = Util.lerp(l, distRange) * heightLengthCoeff;
+    theta += modify;
+    wrist.setToTarget(theta);
+    Logger.recordOutput("VarShootPrime theta", theta);
+    Logger.recordOutput("VarShootPrime modify", modify);
+    Logger.recordOutput("VarShootPrime h", h);
+    Logger.recordOutput("VarShootPrime v", v);
+    Logger.recordOutput("VarShootPrime l", l);
+    Logger.recordOutput(
+        "REAL VALUES PROBABLY", Units.radiansToDegrees(calcTheta(9.81, 2, 1.4, 18)));
+
+    shooter.setPIDReferenceBOTH(v / RPM_MPS);
+  }
+
+  // Source? It was revealed to me by a wise tree in a dream
+  // JK this https://en.wikipedia.org/wiki/Projectile_motion
+  private double calcTheta(double g, double l, double h, double v) {
```

Comment:
Rather than assuming a set velocity, finding the angle needed to hit the x,y coordinate, what if instead we set the x,y coordinate as the apogee, and calculated the velocity required to just get the note in?

2d = v^2sin(2𝜃)/g, where `d` is the distance from the goal
h= v^2sin^2(𝜃)/2g, where h is the height difference between the shooter and the goal

This will give you an angle and a velocity for the trajectory to reach its max height within the goal. We can move the apogee to the back of the speaker to give us some wiggle room. 

It simplifies to 

`theta = atan(2h / d)`
and velocity becomes

`sqrt(2*g*d / sin(2*theta))`

Might be better, might not

Reply:
Definitely want to try this at some point, not now though